proposals/vprim/README.md

```diff
@@ -0,0 +1,300 @@
+`RectLight` and `SphereLight`) or non-boundable (`DomeLight`). Applied
+schema adapters were introduced to make any `Gprim` a light without
+having to redefine in `UsdLux`.
+
+### Example: `VizPoints`
+Let's use points as a motivating example of a `Vprim`. Consider an
+OpenUSD file that contains a mesh reconstruction of data from a point
+cloud scan. A user might benefit from visualizing the points on top of
+the mesh to aid in debugging some errors in the reconstruction. The
+user might choose the `Points` schema to encode positions. And they
+might be able to hide the source point cloud from rendering contexts
+using `purpose` tagging and/or variant sets. But they will be unable
+to make the points all the same size in camera space in a
+specification compliant renderer. Preview optimizations that might
+give a user their desired effect like those found in Storm exist
+primarily for performance reasons and may not persist across
+renderers, sessions, and future releases.
+
+A visualization focused specification of points could look something
+like this--
+
+```
+def VizPoints "PointCloud" {
+    point3d[] points = [...]
+    float[] pixels = [...]
+    color3f[] primvars:displayColor = [(1, 0, 0)]
+}
+```
+
+`VizPoints` could be rendered in camera but may also be composited
+against a depth buffer from a physically based renderer.
+
+`Vprims` would still support primvars and would be primarily styled
+using `displayColor` and `displayOpacity`. We'd like to leave open the
+possibility of a `viz` material purpose as a holder for describing
+nondiegetic styles best described through material network, but this
+proposal doesn't have a concrete definition of this yet.
+
+### Example: `VizOverlayAPI`
+This proposal frames screen space points (and paths) visualization
+primitives as fundamentally different from their `Gprim` counterparts,
+warranting their own first class `Vprim` typed schemas. It's important
+that other `Gprim` types like `Cube` or `Cylinder` that retain their
+underlying definition don't require nondiegetic reflections.
+
+Consider an important primitive that you want to highlight in the
+context of visualization.
+
+```
+def Sphere "grey_sphere" (apiSchemas=["VizOverlayAPI"])
+{
+    double radius = 4
+    color3f primvars:displayColor (0.3, 0.3, 0.3)
+    color3f viz:overlay:color = (0.6, 0.0, 0.0)
+    float viz:overlay:opacity = 0.5
+    token viz:overlay:visability = "inherited"
+}
+```
+This adds a nondiegetic overlay and outline for visualizations to a
+grey sphere. The viewer will see the red transparently overlayed, but
+the sphere is still a grey sphere from the perspective of light
+transport (reflected and refracted light).
+
+Consideration must be also given to if or how these schemas affect
+interactive picking. A non-slicing visualization effect might include
+sillhoutte outlines or overlays applied to a `Gprim` to style or
+highlight its importance.
+
+### Example: `VizScreenAnchor`
```

Comment:
A note on implementation thoughts; I left a bunch of comments on the Autodesk billboards proposal along these lines:

My preferred way to do screen/camera anchoring is to augment the hydra "xform" definition to say that a transform is rooted in view space.  This matches what GlfSimpleLight does for camera lights, and is consistent with the other flag we have in the hydra "xform" definition: resetXformStack, which can be thought of as saying that a transform is rooted at the origin of world space.  The idea is that these xforms would flow to the renderer, and Storm (for example) could turn a view-relative transform into a world-relative transform in the shader, meaning that we don't need to recompute the transforms for each viewer in a multi-viewer app.

If we want to separate position anchoring, rotation anchoring, and scale anchoring, it's a bit more complicated but you can say (for example) that a matrix is translate-anchored to the camera and just compensate the translation as above.

I think it is cool and useful to let transforms below one of these camera-anchoring operations compose, since the composed transform would be (implied inverse view xform) * parent xform * child xform, and the child composed transform just inherits the flag indicating the implied inverse view xform.

This is all consistent with VizScreenAnchor being a scoping mechanism, and would potentially be with an xformOp as well.

Any kind of screen space positioning, scaling, styling is perfectly xformable and boundable in the renderer, of course, but absent a GfFrustum for context the UsdGeomXformCache/UsdGeomBboxCache become meaningless at some point in the tree, so for non-hydra consumers I suppose it's important to flag when that becomes the case.  I think if XformCache/BboxCache had an option to take a GfFrustum, it could replicate the compensation that hydra is doing, and that would be a useful utility to provide.

Reply:
I think consumers like hydra that can guarantee they have a single frustum have a lot of flexibility with how this gets implemented.

In my latest updates, I'm suggesting that transforms and bounds might want to be evaluated with respect to anchoring points-- to most prims that would be the pseudo root and nothing would change, but you'd be able to identify the anchoring point and use whatever extrinsic context was needed (resolution, frustum, etc.).